src/Framework/Traits.cs

```diff
@@ -56,6 +56,11 @@ public Traits()
         /// </summary>
         public readonly bool MSBuildCacheFileEnumerations = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MsBuildCacheFileEnumerations"));
 
+        /// <summary>
+        /// Use Microsoft.Extensions.FileSystemGlobbing for IMSBuildGlob implementation instead of the built-in implementation
+        /// </summary>
+        public readonly bool UseFileSystemGlobbingForMSBuildGlob = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILD_USE_FILESYSTEMGLOBBING"));
```

Comment:
to test with perfstar:
0. cd to your fork
1. `gh pr checkout https://github.com/dotnet/msbuild/pull/12033`
2. `git checkout -b perf/meglobbing-test`
4. ```suggestion
        public readonly bool UseFileSystemGlobbingForMSBuildGlob = true;
```
5. `git add src/Framework/Traits.cs`
6. `git commit -m 'test feature flag'`
7. (optional) `git remote add upstream https://github.com/dotnet/msbuild.git`
8. `git push perf/meglobbing-test:perf/meglobbing-test`